src/plugins/creditor/credNodeStore.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+import {type CreditorNode, type NodeUuid, type NodeTagId} from "./creditorNode";
+import {JsonLog} from "../../util/jsonLog";
+
+export type NodeEntry = {|type: "NODE", entry: CreditorNode|};
+export type NodeDetails = {title: string, id: NodeUuid};
+type NodesFile = JsonLog<NodeEntry>;
+
+export function toNodeEntry(entry: CreditorNode): NodeEntry {
+  return {type: "NODE", entry};
+}
+
+export function fromNodeEntry({entry}: NodeEntry): [NodeUuid, CreditorNode] {
+  return [entry.id, entry];
+}
+
+export function getDetails({id, title}: CreditorNode): NodeDetails {
+  return {id, title};
+}
+
+/**
+ * CredNodeStore is the static, in-memory representation of creditor nodes
+ *
+ * This class is used to accept and validate frontend inputs, and export them in
+ * serialized format utilizing the jsonLog functionality when saving to a file.
+ * Nodes in this class are still modifiable, of course, and jsonLogging is utilized
+ * because it outputs a gitdiff-friendly format
+ *
+ * Nodes in the the cred node store are not deletable at this time. In lieu of deleting nodes,
+ * their weights (`mint` values) can simply be set to zero
+ */
+export default class CredNodeStore {
+  +_nodes: Map<NodeUuid, CreditorNode>;
+
+  constructor(fromDisk: NodesFile | void) {
+    if (!fromDisk) {
+      this._nodes = new Map();
+    } else {
+      const nodeKVArray = Array.from(fromDisk.values()).map(fromNodeEntry);
+      this._nodes = new Map(nodeKVArray);
+    }
+  }
+
+  serialize(): string {
+    const log = new JsonLog();
+    const sortedEntries = Array.from(this.values()).sort(
+      ({createdAt: a}, {createdAt: b}) => a - b
+    );
+    return log.append(sortedEntries.map(toNodeEntry)).toString();
+  }
+
+  set(node: CreditorNode): CredNodeStore {
+    const {
+      id,
+      tags,
+      title,
+      description,
+      graphTimestamp,
+      createdAt,
+      parent,
+    } = node;
+    if (!id) {
+      throw new Error("Creditor Node UUID required");
+    }
+    if (!Array.isArray(tags)) throw new Error("`tags` must be an array");
+    if (!(title || description))
+      throw new Error("`title` and `description` are required in Cred Nodes");
+    if (!(graphTimestamp || createdAt))
+      throw new Error(
+        "`createdAt` and `graphTimestamp` values must be non-zero"
+      );
+    if (parent && !this._nodes.has(parent))
+      throw new Error("cannot add node with nonexistent parent");
+    this._nodes.set(node.id, node);
+    return this;
```

Comment:
We don't usually do runtime validation this extensively, because we assume that Flow will catch issues like the id being unset. However, if we do want to validate, it would be much more comprehensive to use the parser module:

```js
import * as C from "../../util/combo"

const creditorNodeParser: C.Parser<CreditorNode> = C.object({
  id: uuid.parser,
  tags: C.array(uuid.parser),
  title: C.string,
  description: C.string,
  graphTimestamp: C.number,
...
})
```

However, in keeping with the rest of the codebase: I'd say use a parser only when we're actually reading from arbitrary JSON objects, and for here we could just check stuff like validating that the parent exists.